apps/common-app/src/examples/RuntimeTests/tests/animations/withDelay/keepSnapshot.test.tsx

```diff
@@ -0,0 +1,195 @@
+    passiveNative: await updatesContainerNoDelay.getNativeSnapshots(componentPassive),
+  };
+
+  await unmockAnimationTimer();
+  await mockAnimationTimer();
+  await clearRenderOutput();
+  const updatesContainerWithDelay = await recordAnimationUpdates();
+  await render(<TestComponent applyWithDelay={true} testAnimation={testAnimation} delay={delay} />);
+  await wait(350 + delay);
+  componentActive = getTestComponent(TEST_COMPONENT_ACTIVE_REF);
+  componentPassive = getTestComponent(TEST_COMPONENT_PASSIVE_REF);
+
+  const delaySnapshots = {
+    active: updatesContainerWithDelay.getUpdates(componentActive),
+    passive: updatesContainerWithDelay.getUpdates(componentPassive),
+    activeNative: await updatesContainerWithDelay.getNativeSnapshots(componentActive),
+    passiveNative: await updatesContainerWithDelay.getNativeSnapshots(componentPassive),
+  };
+  await clearRenderOutput();
+
+  return {
+    noDelaySnapshots,
+    delaySnapshots,
+  };
+}
+
+function compareActiveAndPassiveSnapshots(
+  msg: string,
+  {
+    active,
+    activeNative,
+    passive,
+    passiveNative,
+  }: Record<'active' | 'activeNative' | 'passive' | 'passiveNative', SingleViewSnapshot>,
+) {
+  // animation from useAnimatedStyle does not record static frames in snapshot
+  const fillerSize = active.length - passive.length;
+  const filler = Array.from({ length: fillerSize }, () => {
+    return {
+      width: 100,
+    };
+  });
+  expect([...filler, ...passive]).toMatchSnapshots(active);
+  expect(active).toMatchNativeSnapshots(activeNative);
+  expect(passive).toMatchNativeSnapshots(passiveNative);
+}
+
+const testCases = [
+  { testAnimation: TestAnimation.TIMING, delay: 0 },
+  { testAnimation: TestAnimation.SEQUENCE, delay: 0 },
+  { testAnimation: TestAnimation.TIMING, delay: 100 },
+  { testAnimation: TestAnimation.TIMING, delay: 1000 },
+  { testAnimation: TestAnimation.SEQUENCE, delay: 100 },
+] as const;
+
+for (const { testAnimation, delay } of testCases) {
+  describe(`Apply **${delay}ms** delay to _${testAnimation}_`, () => {
+    test('Components animated with `withDelay` of have same snapshot but moved in time', async () => {
+      const { delaySnapshots, noDelaySnapshots } = await getSnapshotUpdates(testAnimation, delay);
+
+      compareActiveAndPassiveSnapshots(
+        'Components animated _without_ withDelay in two different ways have matching snapshots',
+        noDelaySnapshots,
+      );
+      compareActiveAndPassiveSnapshots(
+        'Components animated _with_ withDelay in two different ways have matching snapshots',
+        delaySnapshots,
+      );
+      const fillerSize = delaySnapshots.active.length - noDelaySnapshots.active.length;
+      const filler = Array.from({ length: fillerSize }, () => {
+        return {
+          width: 100,
+        };
+      });
+
+      expect([...filler, ...noDelaySnapshots.active]).toMatchSnapshots(delaySnapshots.active);
```

Comment:
I don't understand this logic 🤔

Reply:
I've added a comment explaining that we create some "filler" - a snapshot of static animation, sth like a snapshot of delay.